testing/trino-tests/src/test/java/io/trino/tests/TestLateMaterializationQueries.java

```diff
@@ -41,6 +44,8 @@ protected QueryRunner createQueryRunner()
                 .builder()
                 .amendSession(builder -> builder
                         .setSystemProperty(LATE_MATERIALIZATION, "true")
+                        .setSystemProperty(FILTERING_SEMI_JOIN_TO_INNER, "false")
```

Comment:
That's concerning. We should have end-to-end integration tests for late materialization using defaults for other stuff.
While i can easily understand why you disable join reordering, I am not convinced yet about this one. -- Please add some explanation as a code comment.

Reply:
Added comments.
It will be tested with defaults once it's enabled by default.

Reply:
BTW 
>     public static final String FILTERING_SEMI_JOIN_TO_INNER = "rewrite_filtering_semi_join_to_inner_join";

the constant should have same name as the session toggle name (except for lower/uppercase)

Reply:
thanks for adding a comment